code/game/atoms.dm

```diff
@@ -365,47 +365,192 @@
-	return 1
+/mob/living/carbon/alien/get_blood_dna_list()
+	return list("UNKNOWN DNA" = "X*")
+
+//to add a mob's dna info into an object's blood_DNA list.
+/atom/proc/transfer_mob_blood_dna(mob/living/L)
+	var/new_blood_dna = L.get_blood_dna_list()
+	if(!new_blood_dna)
+		return 0
+	return transfer_blood_dna(new_blood_dna)
 
-/atom/proc/add_blood_list(mob/living/carbon/M)
-	// Returns 0 if we have that blood already
-	if(!istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
+/obj/effect/decal/cleanable/blood/splatter/transfer_mob_blood_dna(mob/living/L)
+	..(L)
+	var/list/b_data = L.get_blood_data(L.get_blood_id())
+	if(b_data)
+		basecolor = b_data["blood_color"]
+	else
+		basecolor = "#A10808"
+	update_icon()
+
+/obj/effect/decal/cleanable/blood/footprints/transfer_mob_blood_dna(mob/living/L)
+	..(L)
+	var/list/b_data = L.get_blood_data(L.get_blood_id())
+	if(b_data)
+		basecolor = b_data["blood_color"]
+	else
+		basecolor = "#A10808"
+	update_icon()
+
+//to add blood dna info to the object's blood_DNA list
+/atom/proc/transfer_blood_dna(list/blood_dna)
+	if(!blood_DNA)
 		blood_DNA = list()
-	//if this blood isn't already in the list, add it
-	if(blood_DNA[M.dna.unique_enzymes])
-		return 0 //already bloodied with this blood. Cannot add more.
-	var/blood_type = "X*"
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		blood_type = H.b_type
-	blood_DNA[M.dna.unique_enzymes] = blood_type
+	var/old_length = blood_DNA.len
+	blood_DNA |= blood_dna
+	if(blood_DNA.len > old_length)
+		return 1//some new blood DNA was added
+
+
+//to add blood from a mob onto something, and transfer their dna info
+/atom/proc/add_mob_blood(mob/living/M)
+	var/list/blood_dna = M.get_blood_dna_list()
+	if(!blood_dna)
+		return 0
+	var/bloodcolor = "#A10808"
+	var/list/b_data = M.get_blood_data(M.get_blood_id())
+	if(b_data)
+		bloodcolor = b_data["blood_color"]
+
+	return add_blood(blood_dna, bloodcolor)
+
+//to add blood onto something, with blood dna info to include.
+/atom/proc/add_blood(list/blood_dna, color)
+	return 0
+
+/obj/add_blood(list/blood_dna, color)
+	return transfer_blood_dna(blood_dna)
+
+/obj/item/add_blood(list/blood_dna, color)
+	var/blood_count = !blood_DNA ? 0 : blood_DNA.len
+	if(!..())
+		return 0
+	if(!blood_count)//apply the blood-splatter overlay if it isn't already in there
+		add_blood_overlay(color)
+	return 1 //we applied blood to the item
+
+/obj/item/clothing/gloves/add_blood(list/blood_dna, color)
+	. = ..()
+	transfer_blood = rand(2, 4)
+
+/turf/add_blood(list/blood_dna, color)
+	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
+	if(!B)
+		B = new /obj/effect/decal/cleanable/blood/splatter(src)
+	B.transfer_blood_dna(blood_dna) //give blood info to the blood decal.
+	B.basecolor = color
+	return 1 //we bloodied the floor
+
+/mob/living/carbon/human/add_blood(list/blood_dna, color)
+	if(wear_suit)
+		wear_suit.add_blood(blood_dna, color)
+		wear_suit.blood_color = color

```

Comment:
Same on all of these `.blood_color = color`- shouldn't `add_blood` take care of this?